### DISCUSSION TITLE
is listener type support NodePort?

### DISCUSSION THREAD

**Discussion by ghost** [emqx/emqx-operator#263] [Q&A] [answered]
<img width="461" alt="image" src="https://user-images.githubusercontent.com/106941947/177745777-e073a88e-d848-4f46-ba39-4bb8025a9aa5.png">

as the titile, is listener type support NodePort?

if support nodePort type listener, 
then can I just configre one port is nodePort, and other port is still ClusterIP type.

example: 
dashboard port 18083 traffic forwarding to NodePort 30803,
and other port is still ClusterIp type.

**Comment by Rory-Z** [comment, marked as answer]
We supported NodePort, but only can use NodePort for all port , or use ClusterIP for all port

**Reply by ghost** [reply]
understand， thanks for your answer.

**Reply by tangxy** [reply]
you can define another service like this：

cat emqx-service.yaml 
apiVersion: v1
kind: Service
metadata:
  name: emqx-export
  namespace: default
  labels:
    "apps.emqx.io/instance": "emqx-export"
spec:
  selector:
    "apps.emqx.io/instance": "emqx"
  ports:
  - name: "mqtt-ws-8083"
    port: 8083
    protocol: "TCP"
    targetPort: 8083
    nodePort: 32683
  - name: "mqtt-wss-8084"
    port: 8084
    protocol: "TCP"
    targetPort: 8084
    nodePort: 32684
  type: NodePort
